C/common/reading_set.cpp

```diff
@@ -192,6 +192,55 @@ ReadingSet::append(const vector<Reading *>& readings)
 	}
 }
 
+/**
+* Deep copy a set of readings to this reading set.
+*/
+bool
+ReadingSet::copy(const ReadingSet& src)
+{
+   vector<Reading *> readings;
+   bool copyResult = true;
+   try
+   {
+	   // Iterate over all the readings in ReadingSet
+	   for ( auto &reading : src.getAllReadings())
+	   {
+		   std::string assetName = reading->getAssetName();
+		   std::vector<Datapoint *> dataPoints;
+		   // Iterate over all the datapoints associated with one reading
+		   for ( auto &dp : reading->getReadingData())
+		   {
+			   std::string dataPointName  = dp->getName();
+			   DatapointValue dv = dp->getData();
+			   dataPoints.emplace_back(new Datapoint(dataPointName, dv));
+			  
+		   }
+		   Reading *in = new Reading(assetName, dataPoints);
```

Comment:
the memory objects created on heap, like "in" and "datapoints members , make sure they are deleted correctly.

Reply:
We are creating new memory to copy the existing datapoints , hence these data points are not supposed to be deleted